### PR TITLE
SRVCOM-1079 Tests for ServiceToService calls

### PR DESCRIPTION
**Pull request by mgencur**
https://issues.redhat.com/browse/SRVCOM-1079

This PR refactors the ServiceMesh tests in that way that it extracts the body of the test which is then reused in two separate table tests: the ServiceMesh related and non-ServiceMesh tests related to service-to-service calls. The ServiceMesh tests still run only if SM is installed. The non-ServiceMesh tests run always.

### CODE REVIEW

**Comment by mgencur** [comment]
@markusthoemmes any chance to review this? thanks

File: test/servinge2e/servicemesh_test.go [hunk 1]
```diff
@@ -156,92 +160,96 @@ func TestKsvcWithServiceMeshSidecar(t *testing.T) {
...
+		})
+	}
+}
 
-			// Let the ksvc be deleted after test
-			caCtx.AddToCleanup(func() error {
-				t.Logf("Cleaning up Knative Service '%s/%s'", service.Namespace, service.Name)
-				return caCtx.Clients.Serving.ServingV1().Services(serviceMeshTestNamespaceName).Delete(context.Background(), service.Name, meta.DeleteOptions{})
-			})
+func testServiceToService(t *testing.T, ctx *test.Context, namespace string, tc testCase) {
```
Comment (nak3) [inline test/servinge2e/servicemesh_test.go:168]: This func should move to `service_to_service_test.go` or some shared place? I think somebody feels strange that `TestServiceToServiceCalls` calls the tests written in `servicemesh_test.go`.

Comment (mgencur) [reply test/servinge2e/servicemesh_test.go:168]: Yeah. Right. Moving to service_to_service_test.go sounds good to me.

**Comment by nak3** [comment]
/lgtm
/hold

holding until test passed just in case.

**Comment by mgencur** [comment]
FWIW, the `hold` is only required if you want to wait for "optional" checks. PRs wait automatically for `Required` checks without the need to use `hold` . And these changes fall within the required checks.

**Comment by nak3** [comment]
/hold cancel

I think upgrade test is completely broken.

```
Running go test with args: -race -count=1 -tags=probe -timeout=30m ./test/upgrade -probe.success_fraction=1.0 --imagetemplate registry.svc.ci.openshift.org/openshift/knative-v0.17.3:knative-serving-test-{{.Name}} --kubeconfig /tmp/tmp.b8VHBZ9wyD --resolvabledomain
Running go test with args: -race -count=1 -tags=probe -timeout=30m ./test/upgrade --pipefile=/tmp/eventing-prober-signal --readyfile=/tmp/eventing-prober-ready --imagetemplate=quay.io/openshift-knative/{{.Name}}:v0.17.2
...# knative.dev/eventing/test/upgrade/prober
test/upgrade/prober/prober.go:37:13: undefined: context
test/upgrade/prober/prober.go:64:34: undefined: ctx
test/upgrade/prober/prober.go:139:8: impossible type assertion:
        *prober does not implement Prober (wrong type for Verify method)
                have Verify() ([]error, int)
                want Verify(<T>) ([]error, int)
WARN invalid TestEvent: FAIL    knative.dev/eventing/test/upgrade [build failed]
bad output from test2json: FAIL knative.dev/eventing/test/upgrade [build failed]

=== Errors
test/upgrade/prober/prober.go:37:13: undefined: context
test/upgrade/prober/prober.go:64:34: undefined: ctx
test/upgrade/prober/prober.go:139:8: impossible type assertion:
        *prober does not implement Prober (wrong type for Verify method)
                have Verify() ([]error, int)
                want Verify(<T>) ([]error, int)
```

/hold cancel

**Pull request merged**